### PR TITLE
refactor(core): Change `abstract get` to `readonly`

### PR DESCRIPTION
**Pull request by tinayuangao**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Remove abstract getters
Issue Number: N/A

## What is the new behavior?
Use readonly variables

## Does this PR introduce a breaking change?
```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: packages/common/src/location/platform_location.ts [hunk 1]
```diff
@@ -34,9 +34,9 @@ export abstract class PlatformLocation {
   abstract onPopState(fn: LocationChangeListener): void;
   abstract onHashChange(fn: LocationChangeListener): void;
 
-  abstract get pathname(): string;
-  abstract get search(): string;
-  abstract get hash(): string;
+  readonly abstract pathname: string;
```
Comment (mhevery) [inline packages/common/src/location/platform_location.ts]: I don't think `abstract` has any meaning in this location. You can drop it.

**Review by mhevery — APPROVED** [review]

File: packages/common/src/location/platform_location.ts (hunk 1 above)
Comment (mhevery) [reply packages/common/src/location/platform_location.ts]: same for all other places

**Comment by mhevery** [comment]
Please address comment, and than add `PR action: merge` label once ready.

**Comment by tinayuangao** [comment]
@mhevery The public api will be different after removing `abstract` : https://travis-ci.org/angular/angular/jobs/280262829
Is it okay to update the public api?

**Review by mhevery — APPROVED** [review]

**Pull request merged**